addons/im_livechat/static/tests/suggestions.test.js

```diff
@@ -65,3 +66,25 @@ test("Cannot mention other channels in a livechat", async () => {
     await insertText(".o-mail-Composer-input", "#");
     await contains(".o-mail-Composer-suggestion", { count: 0 });
 });
+
+test("Internal user mention shows their live chat username", async () => {
+    const pyEnv = await startServer();
+    pyEnv["res.partner"].write([serverState.partnerId], { user_livechat_username: "Batman" });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "livechat",
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId, livechat_member_type: "agent" }),
+            Command.create({
+                partner_id: serverState.publicPartnerId,
+                livechat_member_type: "visitor",
+            }),
+        ],
+    });
+    pyEnv["res.users"]._applyComputesAndValidate();
+    await start();
+    await openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", "@");
+    await click(".o-mail-Composer-suggestion:contains(Mitchell Admin)");
+    await click(".o-mail-Composer button[title='Send']:enabled");
+    await contains(".o-mail-Message a.o_mail_redirect", { text: "@Batman" });
```

Comment:
I feel that the suggestion should have _"Mitchell Admin as Batman"_ so that click on suggestion makes sense that this becomes `@Batman`.
And yes, I think that the composer should contain `@Batman` rather than have it transform on posting message, this is less confusing if suggestion clarifies it and resulting mention in composer is `@Batman`.